File: BOJ/1260/A.py
import sys
from collections import deque
sys.stdin = open("D:\Projects\AA\BOJ\\1260\input.txt")
input = sys.stdin.readline

N, M, V = map(int, input().rstrip().split())


arr = [[] for _ in range(N+1)]
visited = []
for i in range(M):
    a, b = map(int, input().rstrip().split())
    arr[a].append(b)
    arr[b].append(a)

for i in range(len(arr)):
    arr[i].sort()


def DFS(num):
    visited.append(num)
    print(num, end=' ')
    for i in range(len(arr[num])):
        if arr[num][i] in visited:
            continue
        DFS(arr[num][i])


def BFS(num):
    visited.append(num)
    q = deque([num])
    while q:
        c = q.popleft()
        print(c, end=' ')
        for i in range(len(arr[c])):
            if arr[c][i] not in visited:
                q.append(arr[c][i])
                visited.append(arr[c][i])


DFS(V)
print()
visited = []
BFS(V)
print()

File: BOJ/1260/A.cpp
#include <iostream>
#include <vector>
#include <deque>
using namespace std;

vector<int> visited(1001, 0);
vector<int> temp(1001, 0);
vector<vector<int>> arr(1001, temp);

void DFS(int current)
{
    cout << current << " ";
    for (int i = 1; i < 1001; i++)
    {
        if (!arr[current][i] || visited[i])
        {
            continue;
        }
        visited[i] = 1;
        DFS(i);
    }
}

int main()
{
    freopen("input.txt", "rt", stdin);
    cin.tie(0);
    ios::sync_with_stdio(0);
    int N, M, V;
    cin >> N >> M >> V;
    for (int i = 0; i < M; i++)
    {
        int a, b;
        cin >> a >> b;

        arr[a][b] = 1;
        arr[b][a] = 1;
    }
    visited[V] = 1;
    DFS(V);
    cout << "\n";
    deque<int> q;
    visited = vector<int>(1001, 0);
    visited[V] = 1;
    q.push_back(V);
    while (q.size())
    {
        int current = q.front();
        cout << current << " ";
        q.pop_front();
        for (int i = 1; i < 1001; i++)
        {
            if (!arr[current][i] || visited[i])
            {
                continue;
            }
            visited[i] = 1;
            q.push_back(i);
        }
    }
}